### DISCUSSION TITLE
MDT problem :"Error: Invalid type of flag simCodeTarget, expected one of None..."

### DISCUSSION THREAD

**Discussion by wzj7531** [OpenModelica/OpenModelica#12768] [Q&A]
Using MDT in my Eclipse RCP application, after create a helloworld.mo and start simulate, nothing happend, the error is:
"Error: Invalid type of flag simCodeTarget, expected one of None, C, Cpp, omsicpp, ExperimentalEmbeddedC, JavaScript, omsic, XML, MidC but got the string "Dump"."
I have already set omc parameters like:"--interactive=tcp --simCodeTarget=C", but is seems not working. How can I solve this problem？

**Comment by adeas31** [comment]
We need more information. How are you setting the parameters?

The default `simCodeTarget` is `C` so you don't need to set it.

**Reply by wzj7531** [reply]
My app is an Eclipse RCP application，and i have integrated MDT in the app.  The modelica Preference I have configure like this:
 
![D165EB80-8F82-4084-B6B0-5529D9900476](https://github.com/user-attachments/assets/b22b7c45-ba04-402a-bc4b-c891b741adf8)

Select Modelica/Electrical/Analog/Examples/Resistor and click simulate menu , nothing happend. What did i miss？And the eclipse ide console info:
![image](https://github.com/user-attachments/assets/8e783c45-1f39-4e6e-900c-dcdee560f93e)

**Comment by adeas31** [comment]
@adrpo any hint? AFAIK, the command line parameters are only used at the startup.

**Reply by wzj7531** [reply]
All documents related to MDT are based on the integration into the Eclipse SDK 3. x version as an example,I'm not sure if mdt can be integrated into Eclipse RCP applications....

**Comment by wzj7531** [comment]
@adeas31 So i clone mdt repo, add projects to my rcp app, debug.... Then, in the Resistor.mo context menu click simulate,  the SimulateHandler response the simulate event:
![image](https://github.com/user-attachments/assets/9496814a-9c9b-4f0b-9dbc-3185c5975d04)
 It seems like firstElem is not an instance of ImodelicaSourceFile(Actually, firstElem is an instance of InnerClass which extends ModelicaClass), so the fullPath is always null, that's way when click simulate menu, nothing hanppend. 
But why? According to IModelicaSourceFile class comment:
"Represents a Modelica source file (.mo file basically)"
Resistor.mo should be a IModelicaSourceFile instance. Why this happend? And how can i fix it?
